### PR TITLE
Update querier and client for swap claims

### PR DESCRIPTION
**Pull request by rhuairahrighairidh**
This PR is a follow on to #950 adding in the query methods. Merge this one first.

Main changes
- Refactor existing querier and rest/cli code to dedupe it. The rest/cli endpoints are the same, but the rpc endpoints were simplified.
- Restructure `reward-factors` query response so it matches the data more closely. Simplify by removing ability to filter by rewarded denom/collateral type. Don't think anyone uses this endpoint.
- Add swap claims querying to querier
- Refactor the main swap syncing method so you can sync a claim without setting it in the store. Add a `GetSyncedSwapClaim` method used by the querier (and soon the payout methods).

I haven't added tests around the querier. Not great practise but I'm punting on it to get the remaining state machine fixes in before code freeze. Bugs will probably be caught in bugbash.

### CODE REVIEW

File: x/incentive/keeper/rewards_swap.go [hunk 1]
```diff
@@ -92,5 +98,24 @@ func (k Keeper) SynchronizeSwapReward(ctx sdk.Context, poolID string, owner sdk.
 	claim.Reward = claim.Reward.Add(newRewards...)
 	claim.RewardIndexes = claim.RewardIndexes.With(poolID, globalRewardIndexes)
 
-	k.SetSwapClaim(ctx, claim)
+	return claim
+}
+
+// GetSynchronizedSwapClaim fetches a swap claim from the store and syncs rewards for all pools.
+func (k Keeper) GetSynchronizedSwapClaim(ctx sdk.Context, owner sdk.AccAddress) (types.SwapClaim, bool) {
```
Comment (karzak) [inline x/incentive/keeper/rewards_swap.go:105]: +1 this is way cleaner than the previous sync methods, should refactor the others as well when we have time.

**Review by karzak — APPROVED** [review]
Changes look good. As you mention, worth exercising these queries in bugbash/testnet.

File: x/incentive/client/cli/query.go [hunk 2]
```diff
@@ -77,89 +77,51 @@ func queryRewardsCmd(queryRoute string, cdc *codec.Codec) *cobra.Command {
...
-					if err != nil {
-						return err
-					}
+				params := types.NewQueryRewardsParams(page, limit, owner, boolUnsynced)
+				claims, err := executeDelegatorRewardsQuery(queryRoute, cdc, cliCtx, params)
+				if err != nil {
+					return err
+				}
+				return cliCtx.PrintOutput(claims)
+			case "swap":
```
[the remark this answers is not in the document]
Comment (rhuairahrighairidh) [reply x/incentive/client/cli/query.go]: added in https://github.com/Kava-Labs/kava/pull/951/commits/8c957890243e146c6c32f4b272584abe32f57f06

File: x/incentive/keeper/rewards_swap.go (hunk 1 above)
Comment (rhuairahrighairidh) [reply x/incentive/keeper/rewards_swap.go:105]: Hoping to get to it before code freeze

**Pull request merged**